iwyu.cc

```diff
@@ -3623,7 +3630,11 @@ class IwyuAstConsumer
   // Called whenever a variable, function, enum, etc is used.
   bool VisitDeclRefExpr(clang::DeclRefExpr* expr) {
     if (CanIgnoreCurrentASTNode())  return true;
-    ReportDeclUse(CurrentLoc(), expr->getDecl());
+    if (const UsingShadowDecl* found_decl = DynCastFrom(expr->getFoundDecl())) {
+      ReportDeclUse(CurrentLoc(), found_decl->getTargetDecl());
```

Comment:
What about the shadow decl itself? I'm thinking of something like:

    // a.h
    namespace A { void foo(); }

    // b.h
    #include "a.h"
    namespace B { using A::foo; }

    // main.cc
    #include "b.h"
    void use() {
        return B::foo();
    }

It seems like this would record a use of `A::foo` but not `B::foo`? Actually, with this example in mind, it seems more right to just use the decl, regardless of whether it's a shadow decl or not, but I'm probably missing something :-)